### DISCUSSION TITLE
Волна по треку

### DISCUSSION THREAD

**Discussion by BloodyBaRGaIn** [K1llMan/Yandex.Music.Api#40] [Q&A] [answered]
Доброго времени суток! Есть ли возможность получить "волну по треку" средствами реализованных публичных методов библиотеки? И если да, то подскажите, пожалуйста, какой подкласс API необходимо задействовать. Есть метод получения похожих треков, но, наверное, это немного не то 😢.
Заранее спасибо!!!

**Comment by K1llMan** [comment]
Доброго. Моя волна - это одно из радио, идентификатор точный не помню. Можно получить список радио, оттуда найти мою волну. По поводу настройки волны подсказать не могу, скорее всего не реализовано.

**Reply by BloodyBaRGaIn** [reply]
Спасибо за направление! Позже пройдусь по станциям, чтобы понять, какая именно отвечает за волну

**Reply by BloodyBaRGaIn** [reply]
Все нашел, вроде бы все классно!
Еще раз спасибо за подсказку!

**Comment by BloodyBaRGaIn** [comment, marked as answer]
```
public ITrackInfo? GetRadio(string id)
{
    YTrack originTrack = Client.GetTrack(id);
    if (originTrack == null) return null;

    YAlbum? album = originTrack.Albums.FirstOrDefault();
    if (album == null) return null;

    List<YStation> stations = Client.GetRadioStations();
    
    YStation? radio = stations
        .FirstOrDefault(s => string.Equals(
            s.AdParams.GenreName,
            album.Genre,
            StringComparison.OrdinalIgnoreCase));

    if (radio == null) return null;

    List<YSequenceItem> sequence = radio.GetTracks(id);
    if (sequence == null || sequence.Count == 0) return null;

    YSequenceItem? item = sequence.Shuffle().FirstOrDefault();
    if (item == null) return null;

    YTrack next = item.Track;
    if (next == null) return null;

    ITrackInfo track = new YandexTrackInfo(next);
    if (track == null) return null;

    track.Radio = true;

    return track;
}
```

**Reply by BloodyBaRGaIn** [reply]
Попробовал добавить по месту `radio.SendFeedBack(YStationFeedbackType.RadioStarted, originTrack)`, но это бросает исключение об ошибочном запросе YErrorResponse. Было бы здорово понять, что нужно передавать в качестве остальных параметров
